jwst/tests_nightly/general/nirspec/test_nirspec_steps_single.py

```diff
@@ -168,3 +171,107 @@ def test_nrs_fs_single_spec2(self):
+                                           save_results=True)
+
+        # _________________________________________________________________________
+        # Test 1 compare 4 FS 1D extracted spectra from sciene data with
+        # no background added to 4 FS 1D extracted spectra from the output
+        # from MasterBackground subtraction
+
+        # run 1-D extract on results from MasterBackground step
+        result_1d = Extract1dStep.call(result, save_results=True)
+
+        # run 1-D extract on original science data without background
+        input_sci_cal_file = self.get_data(*self.test_dir,
+                                            'nrs1_sci_cal.fits')
+        # find the 1D extraction of this file
+        # find Extract1dStep on sci data to use the same version of
+        # this rountine run on both files  rather than running it off line
+        # and having the 1-D extracted science file stored as input to step
+
+        sci_cal_1d = Extract1dStep.call(input_sci_cal_file, save_results=True)
+
+        # Compare the FS  1D extracted data. These types of data are
+        #  MultiSpec Models
+        # the tolerance on this compares could be the max of the user 1-D
+        user_1d = datamodels.open(input_1dbkg_file)
+        flux_tol = np.amax(user_1d.spec[0].spec_table['flux'])
+
+        num_spec = len(result_1d.spec)
+        atol = flux_tol  # set high for now
+        rtol = 0.05      # set high for now
+        for i in range(num_spec):
+            sci_spec_1d = sci_cal_1d.spec[i].spec_table['flux']
+            # check if we need to pull FLUX or NET from table
+            min_value = sci_spec_1d.min()
+            max_value = sci_spec_1d.max()
+            if min_value == 0 and max_value == 0:
+                sci_spec_1d = sci_cal_1d.spec[i].spec_table['net']
+                result_spec_1d = result_1d.spec[i].spec_table['net']
+            else:
+                result_spec_1d = result_1d.spec[i].spec_table['flux']
+            assert_allclose(sci_spec_1d, result_spec_1d, rtol=rtol, atol=atol)
+        # ______________________________________________________________________
+        # Test 2  compare the science MultiSlit data with no background
+        # to the MultiSlit output from the masterBackground Subtraction step
+        # background subtracted science image. For FS case
+        input_sci_model = datamodels.open(input_sci_cal_file)
+        num_slits = len(input_sci_model.slits)
+
+        for i in range(num_slits):
+            slit_sci = input_sci_model.slits[i].data
+            slit_result = result.slits[i].data
+
+            # check for outliers in the science image - need a method that iterates
+            # do not check for outliers on result - some of those  outliers might be
+            # something we need to flag
+            sci_mean = np.nanmean(slit_sci)
+            sci_std = np.nanstd(slit_sci)
+            upper = sci_mean + sci_std*3.0
+            mask_clean = slit_sci < upper
```

Comment:
Do you want to clip low values as well?

Reply:
Ok I was not sure - I was just thinking of things like cosmic rays that could mess things up.
Should I also clip low vales